src/main/java/org/springframework/data/redis/connection/jedis/JedisClusterConnection.java

```diff
@@ -127,11 +129,51 @@ public JedisClusterConnection(JedisCluster cluster, ClusterCommandExecutor execu
 	 * (non-Javadoc)
 	 * @see org.springframework.data.redis.connection.RedisCommands#execute(java.lang.String, byte[][])
 	 */
+	@Nullable
 	@Override
 	public Object execute(String command, byte[]... args) {
 
-		// TODO: execute command on all nodes? or throw exception requiring to execute command on a specific node
-		throw new UnsupportedOperationException("Execute is currently not supported in cluster mode.");
+		Assert.notNull(command, "Command must not be null!");
+		Assert.notNull(args, "Args must not be null!");
+
+		return clusterCommandExecutor
+				.executeCommandOnArbitraryNode((JedisClusterCommandCallback<Object>) client -> JedisClientUtils.execute(command,
```

Comment:
Hello! @christophstrobl 
I have a question. Why you have chosen `executeCommandOnArbitraryNode`?
What If someone decided to use this API to run another command like:
`connection.execute("FLUSHDB", "ASYNC".getBytes("UTF-8")), true);`
Don't you think that in this case execution behaviour would be different?